### DISCUSSION TITLE
Configuring rules for both .js and .ts

### DISCUSSION THREAD

**Discussion by viveleroi** [eslint/eslint#17498] [Q&A]
Nearly all of our repo files are `.ts` or `.tsx`, however we have some scripts and root configs as `.js` etc.

I originally wrote our eslint config to apply all "non-ts" rules to `.js` files, and then I provided an `override` for all `ts` files. However, it looks like plugins or rules in the `override` block complete replace those defined in the base config.

I want all of my rules to apply to both sets but we have a long list and I don't want to duplicate them. Is there a better approach here or am I misunderstanding how to configure this?

```json

{
  "extends": ["eslint:recommended", "plugin:unicorn/recommended", "plugin:sonarjs/recommended", "prettier"],
  "ignorePatterns": ["!.storybook", "app/vite-env.d.ts", "bin", "build", "coverage", "src", "storybook-static"],
  "plugins": ["@typescript-eslint", "sonarjs"],
  "root": true,
  "rules": {
    // ... a ton of rules go here
    "curly": [2, "all"]
  },
  "overrides": [
    {
      "files": ["playwright.config.ts", "app/**/*.{ts,tsx}", ".storybook/*.ts", "tests/**/*.ts"],
      "plugins": ["eslint-plugin-tsdoc", "sonarjs"],
      "extends": [
        "plugin:react/recommended",
        "plugin:react/jsx-runtime",
        "plugin:react-hooks/recommended",
        "plugin:@typescript-eslint/strict-type-checked",
        "plugin:@typescript-eslint/stylistic-type-checked",
        "plugin:storybook/recommended",
        "plugin:unicorn/recommended",
        "plugin:sonarjs/recommended",
        "prettier"
      ],
      "parser": "@typescript-eslint/parser",
      "parserOptions": {
        "tsconfigRootDir": "./",
        "project": ["./tsconfig.json"]
      },
      "settings": {
        "react": {
          "version": "detect"
        }
      },
      "rules": {
        "@typescript-eslint/consistent-type-exports": 2,
      }
    }
  ]
}
```

**Comment by Rec0iL99** [comment]
Hi @viveleroi, I think all the rules you define in your main config will also be applied in your `overrides` block.

**Reply by viveleroi** [reply]
I'll have to setup a simple test repo but I believe I had to redefine the list of plugins in my override block as it fully overrode the base configs. Maybe that's not the case for rules but seemed to be for plugins.

Because I had to redefine the plugins, I had to redefine some rules too. I'll run some tests.
